### ISSUE TITLE
Parse COMMON blocks as a list of declarations, not expressions

### ISSUE THREAD

**Issue by raehik** [camfort/fortran-src#171]
COMMON blocks are parsed into `StCommon` constructors, which store expressions. But the [Oracle Fortran 77 reference for COMMON](https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vn7v/index.html) states that they should store lists "of variable names, array names, and array declarators". We have `TypeSpec` and `Declarator`s for that. A similar construct, STRUCTUREs, uses them. I think COMMON blocks should have similar behaviour.

Being parsed as expressions means they're affected by the function call/array index disambiguation, which shouldn't happen. This simple snippet:

```fortran
      SUBROUTINE TEST
      COMMON /CMN/ X(2)
      END
```

produces an AST that wraps `X(2)` in a `ExpFunctionCall`. It should actually be an array of size 2.

There's an example program on a tutorial website [here](https://riptutorial.com/fortran/example/11596/common-blocks) that declares an array inside a COMMON block. All the array accesses end up as function calls.

See the later comments on #169 for original discussion.

**Comment by raehik** [comment]
I tried running the example program through CamFort's common block eliminator, but get a "variable is of unknown type" runtime error pointing to the final common block.

**Comment by raehik** [comment]
An [online Fortran tutorial](https://www.tat.physik.uni-tuebingen.de/~kley/lehre/ftn77/tutorial/common.html) suggests *"there is usually nothing to gain by putting arrays in common blocks"*. Maybe this usage isn't common, then? (heh)

**Comment by 0xzhang** [comment]
This usage is really not unusual. But the compiler(gfortran) recognizes that and someone have used :joy:

**Comment by raehik** [comment]
Fixed in #173 . @0xzhang the problematic code you showed before should work fine now:

```fortran
      SUBROUTINE TEST
      INTEGER A
      COMMON /KAN/ A(2)
      END
```

`fortran-src --typecheck issue-173.f`:

```
_test1           - Subroutine
kan_a_common            Integer 4 Array [(Nothing,Just 2)]
```
